src/Slot.js

```diff
@@ -30,13 +30,17 @@ Slot.prototype = {
         });
     },
 
+    wrapInDocumentLoaded: function(code) {
+        return '(function() { var run = function() { ' + code + ' }; if (document.readyState === "loading") { document.addEventListener("DOMContentLoaded", run); } else { run(); } })();';
+    },
+
     buildHtml: function() {
         var output = [];
         for (var i = 0, len = this.content.length; i < len; i++) {
             var content = this.content[i];
             if (content.inline) {
                 if (this.contentType === 'js') {
-                    output.push('<script ...data.inlineScriptAttrs marko-body="static-text">' + content.code + '</script>'); // eslint-disable-line no-template-curly-in-string
+                    output.push('<if(data.externalScriptAttrs && (data.externalScriptAttrs.async || data.externalScriptAttrs.defer))><script ...data.inlineScriptAttrs marko-body="static-text">' + this.wrapInDocumentLoaded(content.code) + '</script></if><else><script ...data.inlineScriptAttrs marko-body="static-text">' + content.code + '</script></else>'); // eslint-disable-line no-template-curly-in-string
```

Comment:
I think it would be slightly cleaner if we moved the `if` to be inside the script block to avoid writing that out twice.

Otherwise this LGTM!